src/test/java/com/google/sps/UserLoginTest.java

```diff
@@ -91,4 +115,47 @@ public void userLoggedOutReturningLogIn() throws ServletException, IOException
     Assert.assertTrue(logInUrl.contains("login"));
     Assert.assertTrue(logOutUrl.isEmpty());
   }
+
+  @Test
+  public void hasProfileReturnsFalse() throws ServletException, IOException  {
+    localHelper.setEnvIsLoggedIn(true);
+
+    User testUser = new User("Bob");
+    String expected = "1234";
+    testUser.setId(expected);
+    String email = "test@example.com";
+    testUser.setEmail(email);
+
+    JsonObject responseJsonObject = getLoginServletResponse();
+
+    String logInUrl = responseJsonObject.get("LogInUrl").getAsString();
+    String logOutUrl = responseJsonObject.get("LogOutUrl").getAsString();
+    Boolean isUserInDatabase = responseJsonObject.get("HasProfile").getAsBoolean();
+    
+    Assert.assertTrue(logOutUrl.contains("logout"));
+    Assert.assertEquals(isUserInDatabase, false);
+  }
+
+  @Test
+  public void hasProfileReturnTrue() throws ServletException, IOException  {
+    localHelper.setEnvIsLoggedIn(true);
+
+    addTestData();
```

Comment:
L143: Maybe have addTestData return the user (e.g.: rename it to "makeAndAddTestUser") it creates rather than recreating it here. In addition to saving space, it makes the test less brittle / reduces maintenance costs (in the present form, you have to change two locations any time you need to update this test).

Reply:
Done